code/frontend/src/components/Answer/AnswerParser.tsx

```diff
@@ -7,23 +7,40 @@ type ParsedAnswer = {
     markdownFormatText: string;
 };
 
+let filteredCitations = [] as Citation[];
+
+// Define a function to check if a citation with the same Chunk_Id already exists in filteredCitations
+const isDuplicate = (citation: Citation,citationIndex:string) => {
+    return filteredCitations.some((c) => c.chunk_id === citation.chunk_id) && !filteredCitations.find((c) => c.id === citationIndex) ;
+};
+
 export function parseAnswer(answer: AskResponse): ParsedAnswer {
     let answerText = answer.answer;
     const citationLinks = answerText.match(/\[(doc\d\d?\d?)]/g);
 
     const lengthDocN = "[doc".length;
 
-    let filteredCitations = [] as Citation[];
+    filteredCitations = [] as Citation[];
     let citationReindex = 0;
+    let i =0
```

Comment:
Is this needed when we lookup the citation index on line 28 already?

Reply:
Not required, checking few other scenario's